lib/bolt/inventory/group.rb

```diff
@@ -107,6 +109,7 @@ def data_merge(data1, data2)
 
         {
           'config' => Bolt::Util.deep_merge(data1['config'], data2['config']),
+          'vars'   => data2['vars'].merge(data1['vars']),
```

Comment:
We made an explicit choice to do a shallow merge here rather than deep merge? That seems sensible to me, might be worth a comment either here or in the vars function (so others can see how inventory collisions are resolved).